build.cake

```diff
@@ -402,6 +404,90 @@ Task("PackageConsole")
         });
     });
 
+Task("PackageChocolatey")
+	.Description("Creates chocolate packages of the console runner")
+    .IsDependentOn("CreateImage")
+	.Does(() =>
+	{
+		var currentImageDir = IMAGE_DIR + "NUnit-" + packageVersion + "/";
+		
+		EnsureDirectoryExists(PACKAGE_DIR);
+		
+		// Note: Since cake does not yet support a working directory and separate output directory for chocolatey, the following copying and hacks are needed.
+		
+		// List with the addins (addin name, primary dll, additional files)
+		var addins = new Tuple<string, string, string[]>[] {
+			new Tuple<string, string, string[]>(
+				"NUnit.Extension.VSProjectLoader",
+				"vs-project-loader.dll",
+				null
+			),
+			new Tuple<string, string, string[]>(
+				"NUnit.Extension.NUnitProjectLoader",
+				"nunit-project-loader.dll",
+				null
+			),
+			new Tuple<string, string, string[]>(
+				"NUnit.Extension.NUnitV2ResultWriter",
+				"nunit-v2-result-writer.dll",
+				null
+			),
+			new Tuple<string, string, string[]>(
+				"NUnit.Extension.NUnitV2Driver",
+				"nunit.v2.driver.dll",
+				new [] {
+					"nunit.core.dll",
+					"nunit.core.interfaces.dll",
+					"nunit.v2.driver.addins"
+				}
+			),
+			new Tuple<string, string, string[]>(
+				"NUnit.Extension.TeamCityEventListener",
+				"teamcity-event-listener.dll",
+				null
+			)
+		};
+		
+		// Install and copy the addin files
+		var toolsDir = "tools";
+		var nugetInstallSettings = new NuGetInstallSettings { OutputDirectory = toolsDir, ExcludeVersion = true };
+		var addinsDir = System.IO.Path.Combine(currentImageDir, "addins");
+		EnsureDirectoryExists(addinsDir);
+		foreach (var addin in addins) {
+			// Install the extension
+			NuGetInstall(addin.Item1, nugetInstallSettings);
+			var addinToolsPath = System.IO.Path.Combine(toolsDir, addin.Item1, "tools");
+			// Copy primary dll
+			var primaryDllPath = System.IO.Path.Combine(addinToolsPath, addin.Item2);
+			CopyFileToDirectory(primaryDllPath, addinsDir);
+			// Copy additional files
+			if (addin.Item3 != null) {
+				foreach (var additionalItem in addin.Item3) {
+					var additionalItemPath = System.IO.Path.Combine(addinToolsPath, additionalItem);
+					CopyFileToDirectory(additionalItemPath, addinsDir);
+				}
+			}
+			// Write the primary dll to the addins file
+			FileAppendLines(System.IO.Path.Combine(currentImageDir, "nunit.engine.addins"), new[] {
```

Comment:
I think the name is OK if it's the __only__  `.addins` file you use. It seems to mean "the .addins file for the engine." But you have been using an approach involving multiple `.addins`, which begs th e question "If this is for the engine, what are the others for.

Another approach might be to follow the approach we use for nuget and call it the `nunit.choco.addins` file. I'm OK with either name though, provided it's the sole file.

See my other comment about getting rid of the other names.

I suggest you don't rely completely on the zip file content, but try to think of the logic that the engine follows in finding its extensions.